components/browser/session/src/main/java/mozilla/components/browser/session/Session.kt

```diff
@@ -238,15 +238,30 @@ class Session(
     /**
      * Set when a load request is received, indicating if the request came from web content, or via a redirect.
      */
-    var loadRequestMetadata: LoadRequestMetadata by Delegates.observable(LoadRequestMetadata.blank) { _, _, new ->
-        notifyObservers {
-            onLoadRequest(
-                this@Session,
-                new.url,
-                new.isSet(LoadRequestOption.REDIRECT),
-                new.isSet(LoadRequestOption.WEB_CONTENT)
-            )
-        }
+    var loadRequestMetadata: Consumable<LoadRequestMetadata>
+        by Delegates.vetoable(Consumable.empty()) { _, _, request ->
+            if (request.peek() == null) {
+                false
+            } else {
+                var consumed: Boolean
+                val new = request.peek() as LoadRequestMetadata
+                val consumers = wrapConsumers<LoadRequestMetadata> {
+                    onLoadRequest(
+                        this@Session,
+                        new.url,
+                        new.isSet(LoadRequestOption.REDIRECT),
+                        new.isSet(LoadRequestOption.WEB_CONTENT)
+                    )
+                }
+
+                consumed = request.consumeBy(consumers)
```

Comment:
Did this work? 🙂 

You can move the `val` down here as well.

Reply:
It did work, but I'll need to test the no feature case.  I'll remove the need review and add it back when I've confirmed everything worked without a app-link feature.  Thanks,

Reply:
I'll move the consumed down as suggested.  Thanks,